ui/apps/profile/src/components/pages/profile-info-page.tsx

```diff
@@ -16,11 +17,21 @@ import {
 import { useGetProfileByDidQuery } from '@akashaorg/ui-awf-hooks/lib/generated/hooks-new';
 import { useParams } from 'react-router';
 import { getProfileImageVersionsWithMediaUrl, useGetLogin } from '@akashaorg/ui-awf-hooks';
+import getSDK from '@akashaorg/awf-sdk';
+import { DIDDocument } from 'did-resolver';
 
 const ProfileInfoPage: React.FC<RootComponentProps> = props => {
   const { plugins } = props;
 
   const { t } = useTranslation('app-profile');
+  const sdk = getSDK();
+
+  async function resolveDid(did: string): Promise<DIDDocument> {
+    return await sdk.services.common.misc.resolveDID(did);
+  }
+
+  const [validDid, setValidDid] = React.useState(false);
+  const [isLoading, setIsLoading] = React.useState(false);
```

Comment:
This along with the corresponding `useEffect` needs to be moved to the hooks folder. You can create hook as follows:

```
export function useValidDid(profileId: string, enabled?: boolean) {
     const [validDid, setValidDid] = React.useState(false);
     const [isLoading, setIsLoading] = React.useState(false);
     
     React.useEffect(() => {
       if(enabled) {
           setIsLoading(true);
           resolveDid(profileId).then(res => {
                setValidDid(Boolean(res));
                setIsLoading(false);
            });
        }
      }, [enabled]);
     
    return enabled ? {validDid, isLoading} : null;
}
```

You can use this latter as `... useValidDid(profileId, !!profileData);`

That make the code more cleaner and readable.